### DISCUSSION TITLE
Failed to connect to server

### DISCUSSION THREAD

**Discussion by lucidMindz** [stitionai/devika#588] [Q&A]
ok so everthing is good on the front end and backend of devika, but when I bring up the browser url for devika, the page loads up fine but a pop up appears saying failed to connect to server and when i go into settings to enter the api keys there is nothing their for me to input the keys.  
Any help is very much appreciated and thank you greatly for your time!

**Comment by SyedMuqtasidAli** [comment]
@lucidMindz i am also facing same issue ui is perfect to load but server connection failed and not allowing to select model or not to prompt. did you solve this issue?

**Reply by lucidMindz** [reply]
No I haven't figured it out yet. I went as far as pip installing Ollama thinking it was as simple as that but I'm still hanging in the wind lol

**Comment by TheBigRigZA** [comment]
Hi, I am also facing the exact same issue, will post if I manage to resolve it, but I think some help might be needed.

**Reply by SyedMuqtasidAli** [reply]
open devika.py file and go to line number 29 and you will see there a URL in double quotes just change from 3000 to 3001 and then rerun the devika server first and then run the uii through this command " bun run start " This will work.

**Reply by TheBigRigZA** [reply]
Thanks, I think this would usually do the trick, but I am battling with some issues to do with Bun, so also ran the ui using npm instead, with your fix in place, but still get an issue. I am running on an Azure Ubuntu instance if that makes a difference. Bun gives this error:

$ vite build && vite preview
vite v5.3.0 building SSR bundle for production...
17894 |   } else {
17895 |     process.stdout.write(output.substring(0, process.stdout.columns - 1));
17896 |   }
17897 | }
17898 | function clearLine$1() {
17899 |   process.stdout.clearLine(0);
        ^
EPERM: Operation not permitted
   errno: -1
 syscall: "write"
      fd: 15

      at write (native:1:1)
      at node:stream:171:43
      at clearLine$1 (/home/azureuser/devika/ui/node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:17899:27)
      at writeLine (/home/azureuser/devika/ui/node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:46740:11)
      at /home/azureuser/devika/ui/node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:46525:617
      at /home/azureuser/devika/ui/node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:46755:17
      at transform (/home/azureuser/devika/ui/node_modules/vite/dist/node/chunks/dep-h78lQ5BT.js:46525:962)
      at /home/azureuser/devika/ui/node_modules/rollup/dist/es/shared/node-entry.js:19774:29

Bun v1.1.12 (Linux x64)
error: script "start" exited with code 1


No Idea, how to fix this, but with NPM it loads seemingly 100%.

Thanks for the tip though!

**Reply by SyedMuqtasidAli** [reply]
@TheBigRigZA check that first the Devika server is in running form, second check npm and node is installed also vite command is installed ?

**Comment by NickMCE** [comment]
I am also facing the same issue, can you please tell me how to solve this

**Comment by TheBigRigZA** [comment]
Hi, thanks, Devika says it's running but I will double check somehow. NPM is installed as is node because NPM is how have been launching the UI successfully with vite.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ichigobranding** [comment]
I am also facing the same issue.
Devika says it's running.UI successfully with vite.

CORS error is the problem?
http://127.0.0.1:1337/api/data => 200
strict-origin-when-cross-origin

"http://127.0.0.1:1337/api/data" is CORS error

![スクリーンショット 2024-06-15 6 46 52](https://github.com/stitionai/devika/assets/4328121/f5036f2e-7127-42b0-bf69-0794a0c6bda3)
![スクリーンショット 2024-06-15 6 47 00](https://github.com/stitionai/devika/assets/4328121/a0ef4def-cf90-461d-ade7-192ee4bcfc8f)

**Reply by SyedMuqtasidAli** [reply]
Elaborate please, you are running on colab and using Apis of running through ollama?

**Reply by ichigobranding** [reply]
@SyedMuqtasidAli 
I'm using Cursor and the OpenAI API.

[API_KEYS]
OPENAI = "sk-xxxx"
[API_ENDPOINTS]
OPENAI = "https://api.openai.com/v1"

Should I use colab and Ollama? My PC is a Mac.

**Reply by SyedMuqtasidAli** [reply]
no no its working fine on mac as well I think the URL you are running is not correct. Give me SS of when you run on first terminal " Devika.py " and on Second Terminal run " bun run start" then give me SS

**Comment by 42zz** [comment]
I encountered with same issue.
added frontend url to this line did work for me.
https://github.com/stitionai/devika/blob/main/devika.py#L32

**Comment by nakeeran-t** [comment]
i am also have the same issue with devika ui run and devika.py also run good but it says failed to connect server

**Reply by SyedMuqtasidAli** [reply]
@if-you-bad-iam-your-dad  https://github.com/stitionai/devika/discussions/588#discussioncomment-9751635 here is solution

**Reply by nakeeran-t** [reply]
i do it but not work @SyedMuqtasidAli

**Reply by SyedMuqtasidAli** [reply]
@if-you-bad-iam-your-dad after make changes rerun server and UI again.

**Reply by nakeeran-t** [reply]
that also not working @SyedMuqtasidAli

**Comment by alexanderbehrens1978** [comment]
I changed in devika.py
COPS(app,....) to COPS(app)
Now it works fine
Try llama2 and google search for the first. It is slow, but it works.

**Reply by nakeeran-t** [reply]
what it mean cops(app...) ?

**Reply by alexanderbehrens1978** [reply]
Wrong Answer... Please correct CORS(app,....) to CORS(app)
It was my fault to write COPS...
